bropkg/src/View/Helper/MarkdownCanonifierHelper.php

```diff
@@ -0,0 +1,62 @@
+<?php
+namespace App\View\Helper;
+
+use Cake\View\Helper;
+
+/**
+ * Markdown canonifier
+ *
+ * This massages Markdown text to make the outcome suitable for rendering
+ * on our site. The expectation is that the output is again transformed
+ * by a markdown renderer -- see Template/Packages/view.ctp.
+ */
+class MarkdownCanonifierHelper extends Helper
+{
+  /**
+   * Canonify Markdown content.
+   *
+   * @param string $input Markdown to be parsed.
+   * @param string $url The package URL for this Markdown.
+   * @return bool|string
+   */
+  public function transform($input, $url)
+  {
+      if (!is_string($input)) {
+          return false;
+      }
+      if (strcasecmp(parse_url($url, PHP_URL_HOST), "github.com") == 0) {
+          /* The package resides on Github. Replace relative links with a Github
+           * blob reference into the source tree. Github does the same, with one
+           * difference: Github knows the default branch, which the blob
+           * reference requires. We do not know that branch, but Github
+           * redirects an unknown branch to the default one. So we assume
+           * "main", and hope for the redirect to work out if needed.
+           */
```

Comment:
This feels sorta bad, in that it could end up with broken links in the text and we wouldn't know unless someone told us. Could we make a request to https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#get-a-repository for the repo and get the `default_branch` value from that?

Reply:
Hrrrm. I already spent way to much time on the  whole thing. Are you thinking of adding that to `bro-pkg-web-updater.php` so it's in the metadata? Seems a bit tricky to do in the webserver.

Reply:
> Are you thinking of adding that to bro-pkg-web-updater.php so it's in the metadata? Seems a bit tricky to do in the webserver.

Yeah, that could be doable. In that regard though, if all of the README data is stored as part of the metadata, is there any reason to do any of this in the web server? Wouldn't it be better to do it one time in the updater and store the fixed version in the metadata?

Reply:
Also, I'm entirely fine with filing a ticket for that and doing it separately, as it would be a bit of rework.